### PR TITLE
Improve printNoMockFoundError message

### PR DESCRIPTION
**Pull request by jnak**
Hello @zth,

This PR improves the error message when no matching mock is found:
- Fix the Jest diff. Expected and received were switched.
- Add specific message when no query was mocked.
- Only show the Jest diff for mocked queries when the operation matches.
- Do not show variables for mocked queries when the operation name does not match.
- Add snapshot tests for all cases.

Btw there is no linter setup so I tried my best to keep my code easy to read. It would be good to set one up at some point to encourage external contributions.

Cheers,
J

### CODE REVIEW

File: __tests__/handleErrors-tests.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
+// @flow
+import { fetchQuery } from '../__testUtils__/fetchQuery';
+import { queryMock } from '../__testUtils__/queryMock';
+
+
+describe('handleErrors', () => {
+  describe('printNoMockFoundError', () => {
+    it('should throw an error when there is no query name', async () => {
+      try {
```
Comment (jnak) [inline __tests__/handleErrors-tests.js:8]: I'm not super happy with this pattern but I could not find a terser way to test exception messages that also caught when errors were not raised.  I tried to use a helper `expectErrorMessageToMatchSnapshot` function but errors would be swallowed as unhandled promises unless expectErrorMessageToMatchSnapshot was itself wrapped with a try / catch. Which defeated most of the purpose of the helper function.

Comment (zth) [reply __tests__/handleErrors-tests.js:8]: Yeah, I feel you... Async/await/promises is a mess when dealing with errors. I think it's just something we'll have to live with for now. I'm fine with it.

File: src/handleErrors.js [hunk 2]
```diff
@@ -9,35 +9,52 @@ export function printNoMockFoundError(
...
+        mockedQueriesMessageArray.push(
+          `Query "variables": ${JSON.stringify(queryMockConfig.variables)}\n` +
+          `Diff of "variables":\n` +
+          `${printDiff(queryMockConfig.variables, variables)}\n\n`
+        );
+    });
+  }
+
+  const otherMockedQueryNames = Object.keys(queryMock._queries).filter(
+    (queryMockName) => (operationName != queryMockName)
```
Comment (zth) [inline src/handleErrors.js]: Mind making this a strict equality check? `!==`.

**Comment by zth** [comment]
This is great stuff! And good catch with the linter, I always run `prettier` myself but it wasn't setup in the repo properly. I've added `prettier` - would you mind merging master again and running `yarn prettier`?

File: src/handleErrors.js (hunk 2 above)
Comment (jnak) [reply src/handleErrors.js]: :thumbsup:

**Comment by jnak** [comment]
Thanks for setting up prettier. Worked like a charm!

PS: I rebased my branch and then force-pushed the changes onto this one. Were there a better way to do this via github PR? We use phabricator at work so I'm not sure.

**Review by zth — APPROVED** [review]

**Pull request merged**